### ISSUE TITLE
No message return from the browser

### ISSUE THREAD

**Issue by meflett** [longvh211/Chromium-Automation-with-CDP-for-VBA#34]
I think the CPD framwork that you have created is exactly what I am looking for but I have run into some issues. 

I am new to the CDP approach but due to changes in the corporate system can no longer use the IE interface on edge. I have downloaded your CDP 2.7.4 and have been trying to get Edge ( the only browser on the system) to work. Running  'runEdge' all the edge instances close and a new one is opened (with a warning tag 'Microsoft Edge is being controlled by automated test software.' in the browser window. Then there is a pause until the time out then an errror is generated. with failure to return message from the browser.
Just wondering if you could give me a clue where the CPD is failing and if there is anything I can do?

Many thanks
M

This is the  generated debug

runedge
13:01:31 | BRID706 | Reattachment aborted as reAttach=True & isLive=False
----------------------------------------------------------------------------------------------------
New Browser Session Started
----------------------------------------------------------------------------------------------------
13:01:32 | BRID706 | cleanUpSessions completed and previous CdpSessionID removed
13:01:32 | BRID706 | setCrashStateNormal completed
13:01:32 | BRID706 | CDPCore initialized successfully
13:02:02 | BRID706 | sendMessage error at line 23: "Timeout waiting for browser to return message."
13:02:02 | BRID706 | strMessage used by sendMessage: {"method":"Target.setDiscoverTargets","params":{"discover":true}, "id":1} 
13:02:02 | BRID706 | attachToSession error at line 1: "sendMessage error at line 23: "Timeout waiting for browser to return message.""
13:02:02 | BRID706 | objAllMessages = {}
13:02:02 | BRID706 | results = 
13:02:02 | BRID706 | results_ = 
13:02:02 | BRID706 | results__ = 
13:02:02 | BRID706 | results___ = 
13:02:02 | BRID706 | .start call string: "C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe" --remote-debugging-pipe --enable-automation --app="http://google.co.uk" 
13:02:02 | BRID706 | ---------------------------------------------------
13:02:02 | BRID706 | BRID706 Information
13:02:02 | BRID706 | ---------------------------------------------------
13:02:02 | BRID706 | Session ID    | sessionID | 
13:02:02 | BRID706 | Last ID       | lngLastID | 2
13:02:02 | BRID706 | Window Handle | brHWnd    | 0
13:02:02 | BRID706 | Window ID     | brWId     | 
13:02:02 | BRID706 | Browser Type  | brType    | msedge.exe
13:02:02 | BRID706 | Browser Ver   | brVer     | 
13:02:02 | BRID706 | Browser Path  | brPath    | C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe
13:02:02 | BRID706 | Tab URL       | brUrl     | 
13:02:02 | BRID706 | Tab Title     | brName    | 
13:02:02 | BRID706 | Tab ID        | targetID  | 
13:02:02 | BRID706 | Core StdInWr  | hStdInWr  | 9996
13:02:02 | BRID706 | Core StdOutRd | hStdOutRd | 10644
13:02:02 | BRID706 | Core CDPInWr  | hCDPInWr  | 10628
13:02:02 | BRID706 | Core CDPOutRd | hCDPOutRd | 10640
13:02:02 | BRID706 | CDP Version   | CDPVer    | 2.7.4

**Comment by longvh211** [comment]
Have you tried any of the demo procedures to see if it works at your end?

**Comment by meflett** [comment]
Yes tried the demo the output is from the "runEdge" demo - although I added the URL for google to check that the browser would launch. 
It then crashes out as below 

13:02:02 | BRID706 | sendMessage error at line 23: "Timeout waiting for browser to return message."
13:02:02 | BRID706 | strMessage used by sendMessage: {"method":"Target.setDiscoverTargets","params":{"discover":true}, "id":1}
13:02:02 | BRID706 | attachToSession error at line 1: "sendMessage error at line 23: "Timeout waiting for browser to return message.""

I'll look at the suggestions link you sent. 

Thanks

**Comment by Jonny125** [comment]
I have the same issue as meflett. All of a sudden the framework stopped to work.
The framework doesn't get a response from the browser and therefore runs in a timeout.
I tried the solution of maxwell by enabling developer mode, however this hasen't solved my issue.

However I noticed it has something to do how edge is modified.
In the comany environment it doesn#t work, however on a private computer it runs just as fine.

Any clues on where to have a look to solve this issue?

**Comment by longvh211** [comment]
Hi Johnny, does your company has Chrome? You can try to see if it works with Chrome.

**Comment by Jonny125** [comment]
Unfortunately we do not use Chrome - and there is no way to get Chrome installed on a corporate notebook.
We have Firefox as an alternative but that won't work.

**Comment by Jonny125** [comment]
After doing some research it seems I have the same issue as meflett, though.
Enabling "Developer Mode" in Edge is not sufficient.
I was able to duplicate the timeout message on my private computer by changing the registry for Microsoft Edge.

The reason for the code to break down is that remote debugging is disallowed.
By changing the following key in the registry
Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Edge\RemoteDebbingAllowed
from 1 to 0 the code will run into timeout.

So please make sure you change it to 1/leave it as 1 or delete it.

As this key is administrated via group policies I need to call IT support to change it.
